addons/link_tracker/tests/test_merge_utm_campaign.py

```diff
@@ -0,0 +1,105 @@
+            'campaign_id': self.utm_campaign_3.id,
+            'medium_id': self.utm_medium.id,
+            'source_id': self.utm_source.id
+        })
+        self.link_tracker_click_1 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_1.id
+        })
+        self.link_tracker_click_2 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_2.id
+        })
+        self.link_tracker_click_3 = self.env['link.tracker.click'].create({
+            'link_id': self.link_tracker_3.id
+        })
+
+    def test_merge(self):
+        """ Merge 2 campaigns and check that everything is correctly updated.
+        Unneeded campaigns should be deactivated, and contain reference to the remaining one.
+        (see 'utm.py'#_merge_utm_campaigns) """
+        (self.utm_campaign_1 + self.utm_campaign_2)._merge_utm_campaigns(self.utm_campaign_1)
+
+        self.assertEqual(self.utm_campaign_2.active, False, 
+            "After merging utm.campaigns, unneeded campaigns should be archived. ")
+        self.assertEqual(self.utm_campaign_2.reference_utm_campaign_id, self.utm_campaign_1, 
+            "After merging utm.campaigns, deactivated utm.campaign should refer to the merged campaign. ")
+
+    def test_merge_unique(self):
+        """ Merge 2 campaigns and check that everything is correctly cleaned up.
+        Link trackers should be cleaned in this case since they're not unique anymore
+        (see 'link_tracker.py'#_clean_duplicates) """
+        link_codes_1 = self.link_tracker_1.link_code_ids
+        link_clicks_1 = self.link_tracker_1.link_click_ids
+        link_codes_2 = self.link_tracker_2.link_code_ids
+        link_clicks_2 = self.link_tracker_2.link_click_ids
+
+        (self.utm_campaign_1 + self.utm_campaign_2)._merge_utm_campaigns(self.utm_campaign_1)
+
+        link_tracker = self.env['link.tracker'].search([('campaign_id', '=', self.utm_campaign_1.id)])
+        link_tracker_empty = self.env['link.tracker'].search([('campaign_id', '=', self.utm_campaign_2.id)])
+
+        self.assertEqual(len(link_tracker.ids), 1, 
+            "After merging utm.campaigns, duplicate link.trackers should be unlinked.")
+        self.assertEqual(len(link_tracker_empty.ids), 0, 
+            "After merging utm.campaigns, no link.tracker link to deactivated utm.campaign.")
+        self.assertFalse(self.link_tracker_2 in self.env['link.tracker'].search([]),
+            "After merging utm.campaigns, duplicate link.trackers should be unlinked.")
+        self.assertEqual(link_tracker.link_code_ids, link_codes_1 + link_codes_2, 
+            "After merging utm.campaigns, link.tracker.code of duplicate link.tracker should be redirected to the remaining one.")
+        self.assertEqual(link_tracker.link_click_ids, link_clicks_1 + link_clicks_2, 
+            "After merging utm.campaigns, link.tracker.click of duplicate link.tracker should be redirected to the remaining one.")
```

Comment:
There is a click_count field on the campaigns.
Can you add some clicks on the trackers and check that the count is correctly summed when merging campaigns 1 and 2 ?

That will make your test even more robust.